### ISSUE TITLE
Migrations review

### ISSUE THREAD

**Issue by bugduino** [AsafSilman/idle-smart-treasury#4]
Hi @AsafSilman 

I gave a brief look at the migrations, I think you should do the migration [number 4](https://github.com/AsafSilman/idle-smart-treasury/blob/main/migrations/4_initialise_bootstrap.js) and [number 5](https://github.com/AsafSilman/idle-smart-treasury/blob/main/migrations/5_initialise_feecollector.js) directly in the constructor of the respective contracts. So eg for the number 4 the constructor of the `SmartTreasuryBootstrap` should accept an `address[]` with the `feeTokens` addresses and then you should call the _registerTokenToDepositList in the constructor directly. The same is valid for the migration number 5 and also for the `transferOwnership` call.

The proposal [number 6](https://github.com/AsafSilman/idle-smart-treasury/blob/main/migrations/6_proposal_phase_1.js) This should be splitted in 2 probably because you have to wait 3 days after the `queue()` call before being able to call the `execute()` 

Will get back later/tomorrow for other migrations and tests which I still need to review

**Comment by AsafSilman** [comment]
Added deposit tokens to constructor with 6d576b01f4c39392f75e1e5f0b96ac13b2f25afa

**Comment by bugduino** [comment]
Ok, other migrations seems correct besides the fact that a copy of the two migrations for the proposals needs (number 6 and number 9) to be made in order to setup the work for mainnet and not only in fork. So you will have eg migration number 6 for both mainnet and mainnet fork

**Comment by bugduino** [comment]
In addition in all migrations you are checking if network is `test` at the very beginning `network === 'test'`, but when running tests against fork, as are currently setup, the network is `development` so all migrations are executed while running tests which is not needed and causes the test suite to take definetly longer than needed. I would add something like this 

``` 
 if (network === 'test' || network === 'development' || network == 'soliditycoverage') {
```

**Comment by AsafSilman** [comment]
Fixed as of acc2a1702af514c7ba63ea69c01c947227cb457c
